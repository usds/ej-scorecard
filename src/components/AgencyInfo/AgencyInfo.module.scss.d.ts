declare namespace AgencyInfoNamespace {
  export interface IAgencyInfoScss {
    agencyInfoCol1: string;
    agencyInfoCol2: string;
    agencyInfoLogo: string;
  }
}

declare const AgencyInfoScssModule: AgencyInfoNamespace.IAgencyInfoScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: AgencyInfoNamespace.IAgencyInfoScss;
};

export = AgencyInfoScssModule;
