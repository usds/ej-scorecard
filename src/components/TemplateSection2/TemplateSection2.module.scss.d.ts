declare namespace TemplateSection2Namespace {
  export interface ITemplateSection2Scss {
    templateSection2Container: string;
  }
}

declare const TemplateSection2ScssModule: TemplateSection2Namespace.ITemplateSection2Scss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: TemplateSection2Namespace.ITemplateSection2Scss;
};

export = TemplateSection2ScssModule;
