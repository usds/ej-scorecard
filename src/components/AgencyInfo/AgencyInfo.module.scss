@use "uswds-core" as *;

.agencyInfoCol1{
    display: flex;
    flex-direction: column;
    align-items: left;
    
};

.agencyInfoCol2{
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    p span {
        display: block;
    }
};

.agencyInfoLogo {
    @include u-margin-left(5);
    @include u-margin-right(5);
    @include u-margin-top(2);
}

.bullet1 {
    @include u-margin-left(4);
    
    li {
        @include u-margin-bottom(2)
    }
}

h1, h2, h3, h4, h5 {
    margin-bottom: unset;
}