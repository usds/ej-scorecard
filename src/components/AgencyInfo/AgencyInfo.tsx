import React from 'react';
import { StaticImage } from 'gatsby-plugin-image';
import { Grid } from '@trussworks/react-uswds';

// import { useAgencyInfo } from '@/hooks/use-agency-info';

import * as styles from './AgencyInfo.module.scss';
import { AgencyInfoProps } from '@/types';
import ScorecardSideNav from '../ScorecardSideNav';

const AgencyInfo: React.FC<AgencyInfoProps> = ({
  info,
  allAgencyNames,
  pathname,
}) => {
  // Todo: may not need this custom hook
  // const agencyInfo = useAgencyInfo();

  return (
    <Grid row gap>
      <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
        <Grid row gap>
          <Grid col={4}>
            <div className={styles.agencyInfoCol1}>
              <StaticImage
                className={styles.agencyInfoLogo}
                src="../../../static/images/agency-a.png"
                alt="Agency logo"
              />
              <h3>{info.Name}</h3>
              <span>{info.Contact_Name}</span>
              <span>{info.Address_Line_1}</span>
              <span>{info.Address_Line_2}</span>
              <span>{info.Phone}</span>
              <span>{info.Site}</span>
            </div>
          </Grid>
          <Grid col={8}>
            <div className={styles.agencyInfoCol2}>
              <h2>{`About ${info.Name}`}</h2>
              <p>{info.About_description}</p>
            </div>
          </Grid>
        </Grid>
      </Grid>
      <Grid desktop={{ col: 1 }} tablet={{ col: 10 }} col={12} />
      <Grid desktop={{ col: 3 }} tablet={{ col: 10 }} col={12}>
        <ScorecardSideNav allAgencyNames={allAgencyNames} pathname={pathname} />
      </Grid>
    </Grid>
  );
};

export default AgencyInfo;
