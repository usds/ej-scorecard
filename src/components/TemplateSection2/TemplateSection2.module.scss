@use "uswds-core" as *;

.templateSection2Container{

}
