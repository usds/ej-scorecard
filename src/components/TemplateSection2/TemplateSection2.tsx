import React from 'react';
import {
  Grid,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading,
} from '@trussworks/react-uswds';

// import * as styles from './TemplateSection2.module.scss';

import { TemplateSectionProps } from '@/types';
import { possessivePlural, splitTextIntoParagraphs } from '../util';
import { DELIMITERS } from '@/data/constants';

const TemplateSection2: React.FC<TemplateSectionProps> = ({ agencyData }) => {
  return (
    <>
      <Grid row gap={6}>
        {agencyData.ECR_HIGH && (
          <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}></Grid>
        )}
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <h2 className={`scorecard-header2`}>
            Environmental and Civil Rights Protection
          </h2>
        </Grid>
      </Grid>
      <Grid row gap={6}>
        {agencyData.ECR_HIGH && (
          <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}>
            <SummaryBox>
              <SummaryBoxHeading headingLevel="h3">
                {`${possessivePlural(agencyData.A_ACRONYM)} Highlights`}
              </SummaryBoxHeading>
              <SummaryBoxContent>
                {agencyData.ECR_HIGH.includes(DELIMITERS.PARAGRAPH)
                  ? splitTextIntoParagraphs(
                      agencyData.ECR_HIGH,
                      DELIMITERS.PARAGRAPH,
                    ).map((paragraph) => paragraph)
                  : agencyData.ECR_HIGH}
              </SummaryBoxContent>
            </SummaryBox>
          </Grid>
        )}
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <p className={`margin-top-0`}>
            Federal agencies are advancing environmental justice by carrying out
            their responsibilities under the law to identify and address
            disproportionate and adverse public health and environmental,
            climate-related, and cumulative impacts on communities with
            environmental justice concerns. Agencies are working to ensure that
            all communities experience the protection of our country’s bedrock
            environmental laws. The following information reflects some of the
            environmental and civil rights work{` `}
            <strong>{agencyData.THE_A_NAME}</strong>
            {` `}
            has done from January 2021 to September 2022.
          </p>

          {agencyData.A_STRATPLANLINK && (
            <p>
              For more information, please visit
              {` `}
              <a
                className="usa-link usa-link--alt usa-link--external"
                href={agencyData.A_STRATPLANLINK}
              >
                {agencyData.A_STRATPLANTXT}
              </a>
              .
            </p>
          )}

          {/* NEPA section is dependent on at least one NEPA metric being defined */}
          {(agencyData.A_NEPA ||
            agencyData.NUM_NEPA ||
            agencyData.NEPA_TRAIN) && (
            <>
              <h3>National Environmental Policy Act</h3>
              <p>
                The National Environmental Policy Act (NEPA) requires federal
                agencies to consider and disclose the environmental, health, and
                community impacts of certain decisions or actions before taking
                them. Environmental reviews help ensure the public is informed
                about potential actions, and give communities an opportunity to
                participate in the government’s decision-making process.
                Environmental reviews also ensure that federal agencies consider
                ways to protect and enhance public health and the environment,
                and advance environmental justice in their decision-making. The
                following information highlights some work{` `}
                <strong>{agencyData.THE_A_NAME}</strong>
                {` `}
                has done from January 2021 to September 2022.
              </p>
              <p>Phase One Scorecard metrics and highlights include:</p>
              <ul className={`bullet1`}>
                {agencyData.A_NEPA && <li>{agencyData.A_NEPA}</li>}
                {agencyData.NUM_NEPA && (
                  <li>
                    <strong>{agencyData.NUM_NEPA}</strong> project
                    modification(s) where environmental justice concerns were
                    raised during the NEPA process
                  </li>
                )}
                <ul>
                  {agencyData.NEPA_H1 && <li>{agencyData.NEPA_H1}</li>}
                  {agencyData.NEPA_H2 && <li>{agencyData.NEPA_H2}</li>}
                  {agencyData.NEPA_H3 && <li>{agencyData.NEPA_H3}</li>}
                </ul>

                {agencyData.NEPA_TRAIN && (
                  <li>
                    <strong>{agencyData.NEPA_TRAIN}</strong> training(s) on
                    environmental justice considerations during NEPA reviews
                  </li>
                )}
              </ul>
            </>
          )}

          {/* TITLEVI section requires at least one TITLEVI metric */}
          {(agencyData.TITLEVI_NAR ||
            agencyData.TITLEVI_N ||
            agencyData.TITLEVI_CR_EJ ||
            agencyData.TITLEVI_R_N) && (
            <>
              <h3>
                Title VI of the Civil Rights Act of 1964 and Environmental
                Justice
              </h3>
              <p>
                Title VI of the Civil Rights Act of 1964 requires that no person
                be excluded from participation in, be denied the benefits of, or
                be subject to discrimination under any program or activity
                receiving federal financial assistance on account of race,
                color, or national origin. Title VI requires federal agencies to
                ensure that programs or activities receiving federal funding,
                including those that affect human health or the environment, do
                not use discriminatory criteria, methods, or practices. The
                following information includes updates from January 2021 to
                September 2022 from <strong>{agencyData.THE_A_NAME}</strong> on
                Title VI that are related to environmental justice.
              </p>
              <p></p>

              <p>Phase One Scorecard metrics and highlights include:</p>
              <ul className={`bullet1`}>
                {agencyData.TITLEVI_NAR && <li>{agencyData.TITLEVI_NAR}</li>}
                {agencyData.TITLEVI_N && (
                  <li>
                    <strong>{agencyData.TITLEVI_N}</strong> open Title VI
                    investigation(s) or compliance review(s) related to
                    environmental justice
                  </li>
                )}
                {agencyData.TITLEVI_R_N && (
                  <li>
                    <strong>{agencyData.TITLEVI_R_N}</strong> closed Title VI
                    compliance review(s) or resolution agreement(s) related to
                    environmental justice
                  </li>
                )}
                {agencyData.TITLEVI_CR_EJ && (
                  <li>
                    <strong>{agencyData.TITLEVI_CR_EJ}</strong> internal
                    training(s) on Title VI civil rights and environmental
                    justice
                  </li>
                )}
              </ul>
            </>
          )}

          {/* Centering Justice section will require all fields from Community Input and Tribal Nations */}
          {(agencyData.A_NAR ||
            agencyData.TA_N ||
            agencyData.PP_N ||
            agencyData.TC_NAR ||
            agencyData.TC_E1 ||
            agencyData.TC_E2 ||
            agencyData.TC_E3) && (
            <>
              <h3>Centering Environmental Justice in Decision-Making</h3>
              <p>
                Federal agencies work to ensure that the voices, perspectives,
                and lived experiences of communities are heard and reflected in
                the priorities, policies, and decision-making of the federal
                government. Agencies also take steps to respect Tribal
                sovereignty and ensure government-to-government consultation on
                federal policies. The following information highlights work{` `}
                <strong>{agencyData.THE_A_NAME}</strong> has done from January
                2021 to September 2022 to center environmental justice in its
                decision-making.
              </p>
            </>
          )}

          {/* Community Input requires at least one metric (A_NAR, TA_N or PP_N) */}
          {(agencyData.A_NAR || agencyData.TA_N || agencyData.PP_N) && (
            <>
              <h4>Community Input and Engagement</h4>
              <p>Phase One Scorecard metrics and highlights include:</p>
            </>
          )}
          <ul className={`bullet1`}>
            {agencyData.A_NAR && <li>{agencyData.A_NAR}</li>}
            {agencyData.TA_N && (
              <li>
                <strong>{agencyData.TA_N}</strong> technical assistance outreach
                event(s) in Fiscal Year 2022
              </li>
            )}
            <ul>
              {agencyData.TA_E1 && <li>{agencyData.TA_E1}</li>}
              {agencyData.TA_E2 && <li>{agencyData.TA_E2}</li>}
              {agencyData.TA_E3 && <li>{agencyData.TA_E3}</li>}
            </ul>
            {agencyData.PP_N && (
              <li>
                <strong>{agencyData.PP_N}</strong> public process(es) related to
                environmental justice, e.g., requests for information or
                listening sessions
              </li>
            )}
            <ul>
              {agencyData.PP_E1 && <li>{agencyData.PP_E1}</li>}
              {agencyData.PP_E2 && <li>{agencyData.PP_E2}</li>}
              {agencyData.PP_E3 && <li>{agencyData.PP_E3}</li>}
            </ul>
          </ul>

          {/* Tribal Nations section requires at least one TC metric */}
          {(agencyData.TC_NAR ||
            agencyData.TC_E1 ||
            agencyData.TC_E2 ||
            agencyData.TC_E3) && (
            <>
              <h4>Consultation and Partnership with Tribal Nations</h4>
              <p>Phase One Scorecard metrics and highlights include:</p>
            </>
          )}
          <ul className={`bullet1`}>
            {agencyData.TC_NAR && <li>{agencyData.TC_NAR}</li>}
            {agencyData.TC_N && (
              <li>
                <strong>{agencyData.TC_N}</strong> Tribal Consultation(s)
                related to environmental justice from January 2021 to September
                2022
              </li>
            )}
            <ul>
              {agencyData.TC_E1 && <li>{agencyData.TC_E1}</li>}
              {agencyData.TC_E2 && <li>{agencyData.TC_E2}</li>}
              {agencyData.TC_E3 && <li>{agencyData.TC_E3}</li>}
            </ul>
            {agencyData.TC_TRAIN_N && (
              <li>
                <strong>{agencyData.TC_TRAIN_N}</strong> internal training(s) on
                Tribal Consultation or working with Tribes and Indigenous
                Peoples to advance environmental justice
              </li>
            )}
          </ul>
        </Grid>
      </Grid>
    </>
  );
};

export default TemplateSection2;
