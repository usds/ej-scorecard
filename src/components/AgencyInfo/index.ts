import AgencyInfo from './AgencyInfo';
export default AgencyInfo;
