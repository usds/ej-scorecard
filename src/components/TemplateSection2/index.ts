import TemplateSection2 from './TemplateSection2';
export default TemplateSection2;
