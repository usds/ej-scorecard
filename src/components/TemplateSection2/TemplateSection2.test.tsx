import createComponentWithIntl from '@/test/testHelpers';
import { agencyData } from '../AgencyInfo/AgencyInfo.test';

import TemplateSection2 from './TemplateSection2';

describe(`rendering of TemplateSection2 Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <TemplateSection2 agencyData={agencyData} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
