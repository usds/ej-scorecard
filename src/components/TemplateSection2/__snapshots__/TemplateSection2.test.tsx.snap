// Vitest Snapshot v1

exports[`rendering of TemplateSection2 Component > checks if component renders 1`] = `
[
  <div
    className="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
      data-testid="grid"
    />
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <h2
        className="scorecard-header2"
      >
        Environmental and Civil Rights Protection
      </h2>
    </div>
  </div>,
  <div
    className="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
      data-testid="grid"
    >
      <div
        className="usa-summary-box"
        data-testid="summary-box"
      >
        <div
          className="usa-summary-box__body"
        >
          <h3
            className="usa-summary-box__heading"
          >
            Environmental Protection Agency Environmental and Civil Rights highlights
          </h3>
          <div
            className="usa-summary-box__text"
          >
            <strong>
              Environmental and Civil Rights highlight space
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <p
        className="margin-top-0"
      >
        Federal agencies are advancing environmental justice by carrying out their responsibilities under the law to identify and address disproportionate and adverse human health, environmental, climate-related and other cumulative impacts on disadvantaged communities, and to ensure that all communities experience the protection of our nation’s bedrock environmental laws. The following information summarizes some of the work the agency has done from January 2021 to September 2022 in these areas.
      </p>
      <h3>
        National Environmental Policy Act
      </h3>
      <p>
        The National Environmental Policy Act (NEPA) requires Federal agencies to consider and disclose the environmental, health, and community impacts of a decision or planned action before taking it. Environmental reviews that agencies conduct through NEPA help ensure the public is informed about the Federal government’s potential actions, that communities have an opportunity to fully and fairly participate in the government’s decision-making process, and that Federal agencies consider ways to protect and enhance the quality of the human environment and advance environmental justice in decision-making.
      </p>
      <ul
        className="bullet1"
      >
        <li>
          <strong>
            EPA outlines NEPA policies and guidelines on their website, including Executive Order 12898, key steps in the NEPA process, how to approach public participation in decision making, and EJScreen—an Environmental Justice mapping and screening tool
          </strong>
        </li>
        <li>
          <strong>
            1
          </strong>
           project modifications where environmental justice concerns were raised during the NEPA process
        </li>
        <ul>
          <li>
            <strong>
              Hosted 3 public meetings with 200 participants regarding projects to reduce Tijuana River watershed pollution that affects communities in Tijuana/San Diego
            </strong>
          </li>
          <li>
            <strong>
              Additional NEPA highlight
            </strong>
          </li>
          <li>
            <strong>
              Additional NEPA highlight
            </strong>
          </li>
        </ul>
        <li>
          <strong>
            421
          </strong>
           trainings on environmental justice considerations during NEPA reviews
        </li>
      </ul>
      <h3>
        Title VI of the Civil Rights Act of 1964 and Environmental Justice
      </h3>
      <p>
        Title VI of the Civil Rights Act of 1964 requires that no person be excluded from participation in, be denied the benefits of, or be subject to discrimination under any program or activity receiving Federal financial assistance on account of their race, color or national origin. Title VI requires Federal agencies to ensure that programs or activities receiving Federal funding, including those that affect human health or the environment, do not use discriminatory criteria, methods, or practices. The following includes updates from 
        <strong>
          Environmental Protection Agency
        </strong>
         on Title VI that are related to environmental justice.
      </p>
      <p>
        <strong>
          Most complaints under investigation involve state and local agencies and allege that actions or inactions have disparate impacts, and have subjected people to discrimination on the basis of race, national origin, disability, and/or age. Complaints involve many types of facilities and services, including: petrochemical, asphalt, concrete, recycling, CAFO, and water treatment and management services/facilities.
        </strong>
      </p>
      <ul
        className="bullet1"
      >
        <li>
          <strong>
            28
          </strong>
           open Title VI or complaints related to environmental justice from January 2021 to September 2022
        </li>
        <li>
          <strong>
            2
          </strong>
           closed Title VI compliance reviews or resolution agreements related to environmental justice from January 2021 to September 2022
        </li>
      </ul>
      <h3>
        Centering Justice in Decision-making
      </h3>
      <p>
        Federal agencies are working to ensure that the voices, perspectives, and lived experiences of environmental justice communities are heard and reflected in the priorities, policies, investments, and decision-making of the Federal government. The following information summarizes some of the work
         
        <strong>
          Environmental Protection Agency
        </strong>
         did from January 2021 to September 2022 to center environmental justice in its decision-making.
      </p>
      <h3>
        Community Input and Engagement
      </h3>
      <ul
        className="bullet1"
      >
        <li>
          <strong>
            Formal partnerships between EPA and non-federal entities helps to advance Environmental Justice and/or inform policy and decision-making. Additionally, cooperative agreements support formal partnership agreements with states, Tribes, and community-based organizations.
          </strong>
        </li>
        <li>
          <strong>
            130
          </strong>
           technical assistance outreach events in Fiscal Year 2022
        </li>
        <ul>
          <li>
            <strong>
              Events/webinars/workshops through The Environmental Justice Thriving Communities Technical Assistance Centers
            </strong>
          </li>
          <li>
            <strong>
              Closing America’s Wastewater Access Gap
            </strong>
          </li>
          <li>
            <strong>
              Technical assistance calls with communities with regional offices
            </strong>
          </li>
        </ul>
        <li>
          <strong>
            480
          </strong>
           public processes related to EJ, e.g. RFIs or listening sessions
        </li>
        <ul>
          <li>
            <strong>
              2 Journey to Justice tours, where EPA Administrator Michael S. Regan heard firsthand from communities facing Environmental Justice concerns in Puerto Rico, Mississippi, Louisiana, and Texas
            </strong>
          </li>
          <li>
            <strong>
              5 public engagement sessions, attended by over 500 people, at 276 Superfund sites where there are environmental concerns. Participants had the opportunity to comment on the Office of Land and Emergency Management Environmental Justice Action Plan.
            </strong>
          </li>
          <li>
            <strong>
              Office of Environmental Justice and External Civil Rights hosted 19 National Environmental Justice Community Engagement calls to inform the public about EPA's Environmental Justice work and enhance opportunities to maintain an open dialogue with advocates.
            </strong>
          </li>
        </ul>
      </ul>
      <h3>
        Consultation and Partnership with Tribal Nations
      </h3>
      <ul
        className="bullet1"
      >
        <li>
          <strong>
            EPA issued the Policy on Environmental Justice for Working with Federally Recognized Tribes and Indigenous Peoples in 2014, to advance Environmental Justice for the protection of the environment and public health
          </strong>
        </li>
        <li>
          <strong>
            18
          </strong>
           Tribal Consultations related to environmental justice from January 2021 to September 2022
        </li>
        <ul>
          <li>
            <strong>
              Formal consultation and coordination with Indian Tribal Governments, as well as public engagement events with communities, on risk management for nine of the first 10 chemicals under risk evaluations by EPA
            </strong>
          </li>
          <li>
            <strong>
              “EJ, Tribes, and Indigenous Peoples” training webinar in May 2022
            </strong>
          </li>
          <li>
            <strong>
              “Reflections on What Indigenous Research Means,” with Dr. Kyle Whyte at the May 2021 Tribal Science Council (TSC) meeting
            </strong>
          </li>
        </ul>
      </ul>
    </div>
  </div>,
]
`;
